eclipse-che/che-operator
# fix: ingress annotation when single-host and gateway exposure

### What does this PR do?

It add ingress annotation `'nginx.ingress.kubernetes.io/proxy-buffer-size: 16k` when deploy eclipse-che with `single-host` and `gateway` exposure. It avoid `502 bad gateway` with long headers provided by keykloack.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?


Fix this issue [#20781](https://github.com/eclipse/che/issues/20781)


### How to test this PR?


I have tested with this steps:
```bash
chectl server:deploy --installer=operator --domain=che-pull-request.rancher-hpd.hm.dm.ad --multiuser --chenamespace=che-pull-request --cheboottimeout=600000 --platform=k8s --che-operator-cr-patch-yaml=che-operator-cr-patch.yaml --postgres-pvc-storage-class-name=nfs-client --workspace-pvc-storage-class-name=nfs-client --che-operator-image=disaster/che-operator:pr-annotation-gateway --skip-kubernetes-health-check
```

che-operator-cr-patch.yaml
```yaml
spec:
  k8s:
    singleHostExposureType: 'gateway'
    tlsSecretName: 'che-tls-certificates'
  metrics:
    enable: false
  server:
    serverExposureStrategy: 'single-host'
    proxyPort: "8080"
    proxyURL: 'http://squid.squid.svc.cluster.local'
    nonProxyHosts: 'localhost|10.0.0.0/8|.hm.dm.ad|.rancher-hpd.hm.dm.ad|.svc.cluster.local'
    useInternalClusterSVCNames: true
    workspaceNamespaceDefault: 'che-<username>'
    #externalDevfileRegistry: true
    #externalPluginRegistry: true
    customCheProperties:
      CHE_LIMITS_USER_WORKSPACES_COUNT: "-1"
      CHE_LIMITS_USER_WORKSPACES_RUN_COUNT: "3"
      CHE_LIMITS_WORKSPACE_IDLE_TIMEOUT: "43200000"
      CHE_INFRA_KUBERNETES_NAMESPACE_ANNOTATIONS: 'che.eclipse.org/username=<username>,field.cattle.io/projectId=c-57d7n:p-8j45h'
      CHE_INFRA_KUBERNETES_NAMESPACE_LABELS: 'app.kubernetes.io/part-of=che.eclipse.org,app.kubernetes.io/component=workspaces-namespace,field.cattle.io/projectId=p-8j45h'
      CHE_WORKSPACE_DEFAULT__CPU__LIMIT__CORES: '1'
      CHE_WORKSPACE_DEFAULT__CPU__REQUEST__CORES: '500m'
      CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES: '300m'
      CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES: '50m'
  storage:
    pvcClaimSize: 4Gi
    pvcStrategy: per-workspace
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
